tensorflow/core/common_runtime/gpu/gpu_cudamallocasync_allocator.cc

```diff
@@ -111,6 +111,32 @@ GpuCudaMallocAsyncAllocator::GpuCudaMallocAsyncAllocator(
         pool_, CU_MEMPOOL_ATTR_REUSE_ALLOW_INTERNAL_DEPENDENCIES, &disable);
   }
 
+  // Set read/write access to all GPUs.
+  for(int i = 0; i < all_pools_.size(); ++i) {
+    // Set the current pool access to the previous GPUs.
+    CUmemAccessDesc map;
+    map.flags = CU_MEM_ACCESS_FLAGS_PROT_READWRITE;
+    map.location.id = all_ids_[i].value();
+
+    map.location.type = CU_MEM_LOCATION_TYPE_DEVICE;
+    if (auto status = cuMemPoolSetAccess(pool_, &map, 1)) {
+      pool_ = nullptr;
+      LOG(FATAL) << "could not set access to the pool : "
+                 << GetCudaErrorMessage(status);
+    }
+
+    // Set the previous pools access to the current GPU.
+    map.location.id = platform_gpu_id.value();
+
+    if (auto status = cuMemPoolSetAccess(*all_pools_[i], &map, 1)) {
+      pool_ = nullptr;
```

Comment:
I think this is a copy-paste error, but even above I'm wondering if we need to set the pool to null.

Reply:
In fact, it is already in a comment: https://github.com/tensorflow/tensorflow/blob/2e6674c890ba925be05588d3daa45e97c7faf6e0/tensorflow/core/common_runtime/gpu/gpu_cudamallocasync_allocator.h#L97-L98